scripts/tasks.js

```diff
@@ -10,20 +12,158 @@ exports.replaceWebpack = () => {
+    originErrorText += ` in ${manifestTypeA}`;
+  }
+  if ((typeof manifestTypeB) === 'string') {
+    if (originErrorText) {
+      originErrorText += ` and ${manifestTypeB}`;
+    } else {
+      originErrorText += ` in ${manifestTypeB}`;
+    }
+  }
+
+  // Deep copy both objects just to be safe
+  const objA = JSON.parse(JSON.stringify(passedObjA));
+  const ret = JSON.parse(JSON.stringify(passedObjB));
+  // Merge A into B
+  // Disabling eslint rule for code clarity, either key exists
+  // in both objects or it doesn't, 2 different cases with subcases
+  /* eslint-disable no-lonely-if */
+  _.forEach(objA, (value, key) => {
+    if (!(key in ret)) {
+      ret[key] = value;
+    } else {
+      // Both objects contain the key
+
+      // It's important we check for Array before Object as Array
+      // is a subset of Object
+      if (isArray(value)) {
+        if (!isArray(ret[key])) {
+          mergeError(`types of key values for key '${key}' \
+don't match when merging${originErrorText}`);
+        }
+        ret[key] = mergeArrays(value, ret[key]);
+      } else if (isObject(value)) {
+        if (!isObject(ret[key]) || isArray(ret[key])) {
+          mergeError(`types of key values for key '${key}' \
+don't match when merging${originErrorText}`);
+        }
+        ret[key] = mergeObjects(value, ret[key], manifestTypeA, manifestTypeB);
+      } else if (key === 'content_security_policy') {
+        // Special case for a special value
+        if ((typeof value) !== 'string' || (typeof ret[key]) !== 'string') {
+          mergeError(`value of content_security_policy was not a string${originErrorText}`);
+        }
+
+        // We first check if there are any duplicate values
+        value.split(';').forEach(rule => {
+          const trimmed = rule.trim();
+          const property = trimmed.split(' ')[0].trim();
+          if (property && ret[key].indexOf(property) >= 0) {
+            mergeError(`Duplicate properties attempting to be \
+merged in content_security_policy${originErrorText}`);
+          }
+        });
+        ret[key] += ` ${value}`;
+      } else {
+        mergeError(`Trying to merge non-objects for key '${key}' \
+in ${manifestTypeA} and ${manifestTypeB}`);
+      }
+    }
+  });
+  /* eslint-enable no-lonely-if */
+  return ret;
+}
+
+
+function mergeManifests(manifestTypeA, manifestTypeB) {
+  let jsonA;
+  let jsonB;
+  if ((typeof manifestTypeA) === 'string') {
+    jsonA = JSON.parse(fs.readFileSync(`${__dirname}/../chrome/manifest.${manifestTypeA}.json`));
+  } else {
+    // Deep copy to be on the safe side
+    jsonA = JSON.parse(JSON.stringify(manifestTypeA));
+  }
+
+  if ((typeof manifestTypeB) === 'string') {
+    jsonB = JSON.parse(fs.readFileSync(`${__dirname}/../chrome/manifest.${manifestTypeB}.json`));
+  } else {
+    // Deep copy to be on the safe side
+    jsonB = JSON.parse(JSON.stringify(manifestTypeB));
+  }
+  return mergeObjects(jsonA, jsonB, manifestTypeA, manifestTypeB);
+}
+
```

Comment:
Wow. This entire green block is some really deep code. It's complicated and troubleshooting any errors or extending it in the future will be intense. It seems that the majority of what this does over built-in functions like `Object.assign()` is that this outputs a ton of error messages. Do we really need all of this fanciness and complexity?